src/field_theory/splitting_field.lean

```diff
@@ -44,103 +45,84 @@ variables {F : Type u} {K : Type v} {L : Type w}
 
 namespace polynomial
 
-variables [field K] [field L] [field F]
 open polynomial
 
 section splits
 
+section comm_ring
+variables [comm_ring K] [field L] [field F]
 variables (i : K →+* L)
 
 /-- A polynomial `splits` iff it is zero or all of its irreducible factors have `degree` 1. -/
 def splits (f : K[X]) : Prop :=
-f = 0 ∨ ∀ {g : L[X]}, irreducible g → g ∣ f.map i → degree g = 1
+f.map i = 0 ∨ ∀ {g : L[X]}, irreducible g → g ∣ f.map i → degree g = 1
```

Comment:
I think for general commutative rings the definition of `splits` should say it's a product of linear factors, which is  only equivalent to this condition when L is a domain; ideally, we'd like to be able to formalize https://mathoverflow.net/a/123526/3332.

Reply:
I agree that it would be a big refactor. So I just relax the typeclass of K.

Reply:
Oh I see, you didn't do anything to L.